lib/assert.js

```diff
@@ -330,7 +330,14 @@ function _throws(shouldThrow, block, expected, message) {
     fail(actual, expected, 'Missing expected exception' + message);
   }
 
-  if (!shouldThrow && expectedException(actual, expected)) {
+  const userProvidedMessage = typeof message === 'string';
```

Comment:
A year late to the party here but won't this _always_ evaluate to `true` since `message` is forced into a string 7 lines up (line 326)?

ping @diversario, @jasnell